### PR TITLE
chore/SOF-6428: build dependency tree

### PR DESCRIPTION
**Pull request by spectre007**
This PR concerns an automatic way of building the dependency tree for RJSF schemas based on a given model subtree and a given schema reference.

### CODE REVIEW

File: src/utils/schemas.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import lodash from "lodash";
```
Comment (k0stik) [inline src/utils/schemas.js:1]: It's better to load only lodash functions that we actually need in the module, not a whole lodash library.
https://stackoverflow.com/questions/35250500/correct-way-to-import-lodash

Comment (spectre007) [reply src/utils/schemas.js]: good point! I adjusted it for all `lodash` imports in the repo

File: src/context/provider.js [hunk 2]
```diff
@@ -9,7 +9,8 @@
  *             to next one, for example data about material to track when it is changed.
  * @notes   Should hold static data only (see `setData` method), no classes or functions
  */
-import lodash from "lodash";
+import capitalize from "lodash/capitalize";
```
Comment (timurbazhirov) [inline src/context/provider.js:12]: I understand where you're coming from on this, but I have to go with readability on this: `lodash.get` is way more understandable than `get` when reading the code. Plus, the 150.5KB size is not as important for us, given that the bundle size is measured in MBs. You can put a separate task for it, if you'd like.

File: src/utils/schemas.js [hunk 3]
```diff
@@ -19,3 +24,103 @@ export function getSchemaByClassName(className) {
...
+        enumNames: nodes.map((node) => get(node, node.dataSelector.name)),
+    };
+}
+
+/**
+ * @summary Recursively generate `dependencies` for RJSF schema based on tree.
+ * @param {Object[]} nodes - Array of nodes (e.g. `[tree]` or `node.children`)
+ * @returns {{}|{dependencies: {}}}
+ */
+export function buildDependencies(nodes) {
```
Comment (timurbazhirov) [inline src/utils/schemas.js:59]: buildRJSFSchemaDependencies

File: src/utils/schemas.js [hunk 4]
```diff
@@ -19,3 +24,103 @@ export function getSchemaByClassName(className) {
...
+
+    // RJSF does not automatically render dropdown widget if `enum` is not present
+    if (modifyProperties && nodes.length) {
+        const mod = {
+            [nodes[0].dataSelector.key]: {
+                ...getEnumNames(nodes),
+                ...getEnumValues(nodes),
+            },
+        };
+        forEach(mod, (extraFields, key) => {
```
Comment (timurbazhirov) [inline src/utils/schemas.js:115]: Why not use the built-in Array.forEach?

File: src/context/provider.js (hunk 2 above)
Comment (spectre007) [reply src/context/provider.js:12]: ok no problem, I can change it back

File: src/utils/schemas.js (hunk 4 above)
Comment (spectre007) [reply src/utils/schemas.js:115]: This is due to calling `forEach` on an object. I can also implement it using `Array.forEach` where `mod` is a list of objects.

File: src/utils/schemas.js (hunk 1 above)
Comment (timurbazhirov) [reply src/utils/schemas.js:1]: @k0stik - the idea is good, provided that readability doesn't suffer, but let's deal with it in a separate task, pls see https://github.com/Exabyte-io/code.js/pull/40#discussion_r1089584173

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**